support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/web/flow/action/DetermineDuoUserAccountAction.java

```diff
@@ -29,60 +25,41 @@
 @RequiredArgsConstructor
 @Slf4j
 public class DetermineDuoUserAccountAction extends AbstractAction {
-    private final VariegatedMultifactorAuthenticationProvider provider;
-    private final ApplicationContext applicationContext;
 
     @Override
-    public Event doExecute(final RequestContext requestContext) {
+    protected Event doExecute(final RequestContext requestContext) {
         val authentication = WebUtils.getAuthentication(requestContext);
         val principal = authentication.getPrincipal();
-        val service = WebUtils.getRegisteredService(requestContext);
+        val flowId = requestContext.getActiveFlow().getId();
+        val applicationContext = ApplicationContextProvider.getApplicationContext();
+        val provider = (DuoMultifactorAuthenticationProvider)
+                MultifactorAuthenticationUtils.getMultifactorAuthenticationProviderById(flowId,
+                        applicationContext).orElseThrow(AuthenticationException::new);
 
         val enrollEvent = new EventFactorySupport().event(this, CasWebflowConstants.TRANSITION_ID_ENROLL);
         val denyEvent = new EventFactorySupport().event(this, CasWebflowConstants.TRANSITION_ID_DENY);
         val unavailableEvent = new EventFactorySupport().event(this, CasWebflowConstants.TRANSITION_ID_UNAVAILABLE);
         val errorEvent = new EventFactorySupport().event(this, CasWebflowConstants.TRANSITION_ID_ERROR);
 
-        val providerIds = WebUtils.getResolvedMultifactorAuthenticationProviders(requestContext);
-        val providers = MultifactorAuthenticationUtils.getMultifactorAuthenticationProvidersByIds(providerIds, applicationContext);
-
-        for (val pr : providers) {
-            val duoProvider = this.provider.findProvider(pr.getId(), DuoMultifactorAuthenticationProvider.class);
-            val duoAuthenticationService = duoProvider.getDuoAuthenticationService();
-            val account = duoAuthenticationService.getDuoUserAccount(principal.getId());
-            LOGGER.debug("Duo user account status is determined as [{}]", account);
-
-            if (account.getStatus() == DuoUserAccountAuthStatus.DENY) {
-                return denyEvent;
-            }
-
-            if (account.getStatus() == DuoUserAccountAuthStatus.ENROLL) {
-                if (StringUtils.isNotBlank(duoProvider.getRegistrationUrl())) {
+        val duoAuthenticationService = provider.getDuoAuthenticationService();
+        val account = duoAuthenticationService.getDuoUserAccount(principal.getId());
+        switch (account.getStatus()) {
```

Comment:
You are better off using if statements here. An empty default and a case that returns just look weird.